src/radical/pilot/session.py

```diff
@@ -626,16 +629,26 @@ def add_resource_config(self, resource_config):
 
         if isinstance(resource_config, str):
 
-            # let exceptions fall through
             rcs = ru.Config('radical.pilot.resource', name=resource_config)
+            domain = os.path.splitext(os.path.basename(resource_config))[0]
+            if domain not in self._rcfgs:
+                self._rcfgs[domain] = {}
 
             for rc in rcs:
-                self._log.info('load rcfg for %s' % rc)
-                self._rcfgs[rc] = rcs[rc].as_dict()
+                self._log.info('load rcfg for "%s.%s"' % (domain, rc))
+                self._rcfgs[domain][rc] = rcs[rc].as_dict()
 
         else:
-            self._log.debug('load rcfg for %s', resource_config.label)
-            self._rcfgs[resource_config.label] = resource_config.as_dict()
+
+            if not resource_config.label:
+                resource_config.label = RESOURCE_CONFIG_LABEL_DEFAULT
+
+            if '.' in resource_config.label:
+                domain, host = resource_config.label.split('.', 1)
+            else:
+                domain = host = resource_config.label
```

Comment:
Same here - I would avoid confusion of changing the label: the user would then need to understand that `mycluster` will change to `mycluster.mycluster`, which is not really obvious.  I would suggest to instead expect the site (domain) and host to be specified.

Reply:
Ok, I would agree. The format of a label should be enforced. I'll add an exception